erpnext/stock/report/stock_balance/stock_balance.js

```diff
@@ -49,7 +49,22 @@ frappe.query_reports["Stock Balance"] = {
 			"label": __("Warehouse"),
 			"fieldtype": "Link",
 			"width": "80",
-			"options": "Warehouse"
+			"options": "Warehouse",
+			get_query: () => {
```

Comment:
Filter by warehouse type only if type is selected, otherwise show all warehouses.

Reply:
done